src/manage_nsfs/manage_nsfs_validations.js

```diff
@@ -445,6 +447,15 @@ async function validate_bucket_args(config_fs, data, action) {
             }
         }
     }
+
+    //if there's a change to the bucket's notifications, we need to test them
+    //if one of the specified notifications fail, we need to fail the user's request
+    if (user_input.notifications) {
+        const test_notif_err = await notifications_util.test_notifications(user_input.notifications, config_fs.connections_dir_path);
+        if (test_notif_err) {
+            throw_cli_error(ManageCLIError.InvalidArgument, "Failed to update notifications", test_notif_err);
```

Comment:
Could you move this part to a separate function? (the function `validate_bucket_args` and then called it (the function has many lines before your change).